File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    let numer = (numer1 * denom2) + (numer2 * denom1);
    let denom = denom1 * denom2;
    
    let divArr = [];
    let min = numer < denom ? numer : denom;
    
    for (let i = 1; i <= min; i++) {
        if (min % i === 0) {
            divArr.push(i);
        }
    }
    
    divArr.sort((a, b) => b - a);
    
    for (let item of divArr) {
        if (numer % item === 0 && denom % item === 0) {
            numer /= item;
            denom /= item;
        }
    }
    
    return [numer, denom];
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import math

def solution(numer1, denom1, numer2, denom2):
    answer = []
    answer_numer = 0
    answer_denom = 0
    
    if denom1 == denom2:
        answer_numer = numer1 + numer2
        answer_denom = denom1
    else:
        answer_denom = denom1 * denom2
        answer_numer = (numer1 * denom2) + (numer2 * denom1)
    
    g = math.gcd(answer_numer, answer_denom)
    
    answer.append(answer_numer / g)
    answer.append(answer_denom / g)
    
    return answer